### ISSUE TITLE
Form with url option -> auth security error

### ISSUE THREAD

**Issue by wvdongen** [cakephp/cakephp#7663]
I’m my quest to figure out how I can post a form to a different controller action without disabling the security options for that action I found some strange behavior.

For example, I’ve got a login action in my users controller. The security component is enabled. In my view users/login.ctp I tested two options to start the form. When I start my form with ’$this->Form->create('User')’ the form works as expected. However when I use $this->Form->create('User', ['url' => ['controller' => 'users', 'action' => 'login']] I get a black hole ‘auth’ error when posting the form. The form should still pass the Security form validation using the 'url' option. This way I can never post a form to a different controller with the Security validatePost setting enabled.

Using cake 2

**Comment by markstory** [comment]
Does the 'different' controller also load SecurityComponent? It must do that to ensure that tokens are correctly generated.

**Comment by wvdongen** [comment]
Yes, the security component is loaded in the other controller.

I started testing with the same controller, action and view as i figured that should at least work. But setting the target with the URL option pointing to the same controller and action shows that a different token is generated than leaving it empty (both methods generatie exactly the same form with the same target). The form with the target set by using the Url option does not pass the filtering of the security component.

**Comment by markstory** [comment]
OK, I can to see if I can reproduce the issue.

**Comment by markstory** [comment]
I was not able to reproduce this issue. I used two controllers - ArticlesController & TasksController. Both controllers had `public $components = ['Security'];` I tested the following scenarios:

* Create a form on `/articles/other` that posts into `/tasks/add`.
* Create a form on `/tasks/add` that posts into `/articles/other`.
* Create a form on `/tasks/add` that posts to the same controller action.

In all three scenarios, I did not see a blackhole exception. I'll need a way to reproduce the issue you're having to continue looking at this.
